addons/stock/models/stock_location.py

```diff
@@ -152,11 +152,12 @@ def _compute_warehouse_id(self):
         view_by_wh = OrderedDict((wh.view_location_id.id, wh.id) for wh in warehouses)
         self.warehouse_id = False
         for loc in self:
-            path = set(int(loc_id) for loc_id in loc.parent_path.split('/')[:-1])
-            for view_location_id in view_by_wh:
-                if view_location_id in path:
-                    loc.warehouse_id = view_by_wh[view_location_id]
-                    break
+            if loc.parent_path:
```

Comment:
To avoid lot of indentations, inverse the condition and use `continue` keyword (It is more readable).
```python
        for loc in self:
              if not loc.parent_path:
                    continue
              path = set(int(loc_id) for loc_id in loc.parent_path.split('/')[:-1])
              ...
```
Or use filtered on self.